pinot-common/src/main/java/org/apache/pinot/common/function/scalar/CaseWhenScalarFunction.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar;
+
+import javax.annotation.Nullable;
+import org.apache.pinot.common.function.FunctionInfo;
+import org.apache.pinot.common.function.PinotScalarFunction;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+
+@ScalarFunction(nullableParameters = true, names = {"case", "caseWhen"})
+public class CaseWhenScalarFunction implements PinotScalarFunction {
```

Comment:
Do we need to have a class for it? Seems it doesn't involve polymorphism, so `isVarArg` should be able to handle it. See `ArrayFunctions.arrayValueConstructor` as an example

Reply:
Will the registered `SqlStdOperatorTable.CASE` perform this type matching? Ideally the type matching should happen in the `SqlOperator` instead of here